moped-editor/src/components/GridTable/FiltersChips.js

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import { Box, Typography, Chip, Grid } from "@mui/material";
+import makeStyles from "@mui/styles/makeStyles";
+
+const useStyles = makeStyles((theme) => ({
+  filtersList: {
+    paddingTop: theme.spacing(1),
+    marginRight: "12px",
+    display: "flex",
+    alignItems: "start",
+  },
+  filtersText: {
+    fontFamily: "Roboto",
+    fontSize: ".9rem",
+    color: theme.palette.text.secondary,
+  },
+  filtersSpan: {
+    fontWeight: 600,
+    textTransform: "uppercase",
+  },
+}));
+
+/**
+ * Create text to show advanced filters and logic applied in the UI
+ * @param {Object} filters - The current filters applied
+ * @param {Boolean} isOr - The current logic applied
+ * @returns {string} - The text to display
+ */
+const makeFilteredByText = (filters, isOr) => {
+  const filtersCount = Object.keys(filters).length;
+  let filteredByText = "Filtered by ";
+
+  /* Only show logic string if more than one filter applied */
+  if (filtersCount === 1) return filteredByText;
+
+  if (isOr) {
+    filteredByText += "any ";
+  } else {
+    filteredByText += "all ";
+  }
+
+  return filteredByText;
+};
+
+/**
+ * Renders filters applied in advanced search
+ * @param {Object} filtersConfig - The filters configuration for the current table
+ * @param {Array|Objects} filters - applied filters
+ * @param {Boolean} isOr - true if ANY filters are matched, false if ALL
+ * @return {JSX.Element}
+ * @constructor
+ */
+const FiltersChips = ({ filters, isOr, filtersConfig }) => {
+  const classes = useStyles();
+
+  const filtersApplied = filters.map((filter) => {
+    const fieldFilterConfig = filtersConfig.fields.find(
```

Comment:
this is a good point and something we've talked about in PRs before too. i'm going to create a tech debt issue to capture this so it doesn't get away from it again. `.find` on an array inside of a .`map` stands out to me too as something that won't scale well and it would be nice to have an config object waiting for a key here. 🙏